core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4548,29 +4548,42 @@ public Object getTarget() {
         try {
             checkPermission(READ);
         } catch (AccessDeniedException e) {
-            String rest = Stapler.getCurrentRequest().getRestOfPath();
-            for (String name : ALWAYS_READABLE_PATHS) {
-                if (rest.startsWith(name)) {
-                    return this;
-                }
-            }
-            for (String name : getUnprotectedRootActions()) {
-                if (rest.startsWith("/" + name + "/") || rest.equals("/" + name)) {
-                    return this;
-                }
-            }
-
-            // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-            if (rest.matches("/computer/[^/]+/slave-agent[.]jnlp")
-                && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
+            if (isPathUnprotected(Stapler.getCurrentRequest().getRestOfPath())) {
                 return this;
             }
 
-
             throw e;
         }
         return this;
     }
+    
+    /**
+     * Test a path to see if it does not require authentication
```

Comment:
Okay, so I'm not understanding something properly then, and may need a better explanation.

`isSubjectToMandatoryReadPermissionCheck` (to me) implies that the function would return true for any path that should go through the standard Jenkins permissions checks, which would be the opposite of the logic currently in this pull request. That is why I thought you might have meant `isNotSubjectToMandatoryReadPermissionCheck`, since to the understanding I have had, that would match the current logic.